### PR TITLE
Change fetch all queries

### PR DESCRIPTION
**Pull request by guavajellyaaron**
- Fixes: 68
- Fixes: 69

### CODE REVIEW

File: netlify/helpers/firebase.ts [hunk 1]
```diff
@@ -82,33 +82,49 @@ export const getReservedHandles = async (): Promise<ReservedHandlesType | false>
...
     });
 }
 
-export const getPaidSessions = async (): Promise<ActiveSessionType[] | false> => {
+export const getActiveSessionsByHandle = async (handle: string): Promise<ActiveSessionType | null> => {
+  return firebase
+    .firestore()
+    .collection(buildCollectionNameWithSuffix("/activeSessions"))
+    .where("handle", "==", handle)
+    .get()
```
Comment (cjkoepke) [inline netlify/helpers/firebase.ts:106]: @guavajellyaaron Do we want to force a single result here? i.e. `.limit(1)`

File: netlify/helpers/firebase.ts [hunk 2]
```diff
@@ -82,33 +82,49 @@ export const getReservedHandles = async (): Promise<ReservedHandlesType | false>
...
+
+      return snapshot?.docs[0]?.data() as ActiveSessionType;
+    });
+}
+
+export const getPaidSessionByHandle = async (handle: string): Promise<ActiveSessionType | null> => {
   return firebase
     .firestore()
     .collection(buildCollectionNameWithSuffix("/paidSessions"))
+    .where("handle", "==", handle).limit(1)
```
Comment (cjkoepke) [inline netlify/helpers/firebase.ts:120]: @guavajellyaaron same here, `.limit(1)`

**Review by cjkoepke — APPROVED** [review]
@guavajellyaaron minor comments, MWR!

File: netlify/helpers/firebase.ts [hunk 3]
```diff
@@ -82,33 +82,49 @@ export const getReservedHandles = async (): Promise<ReservedHandlesType | false>
...
+    .firestore()
+    .collection(buildCollectionNameWithSuffix("/activeSessions"))
+    .where("handle", "==", handle)
+    .get()
+    .then(snapshot => {
+      if (snapshot.empty) {
+        return null;
+      }
+
+      return snapshot?.docs[0]?.data() as ActiveSessionType;
```
Comment (cjkoepke) [inline netlify/helpers/firebase.ts:112]: @guavajellyaaron We're only returning the first result here. Shouldn't it be all of them?

**Review by cjkoepke — CHANGES_REQUESTED** [review]

File: netlify/helpers/firebase.ts (hunk 3 above)
Comment (guavajellyaaron) [reply netlify/helpers/firebase.ts:112]: @cjkoepke This function was updated to find a matching sessions for a particular handle or phone number. We're only checking if it exists.

Comment (cjkoepke) [reply netlify/helpers/firebase.ts:112]: @guavajellyaaron Ah, I see. Let's change the function name then to be singular.

**Review by cjkoepke — APPROVED** [review]
change function name but MWR

File: netlify/helpers/firebase.ts (hunk 3 above)
Comment (guavajellyaaron) [reply netlify/helpers/firebase.ts:112]: @cjkoepke Good call!

**Review by cjkoepke — APPROVED** [review]
LGTM

**Pull request merged**